### PR TITLE
PLN

### PR DESCRIPTION
**Pull request by suprajasridhara**
Pull request for boilerplate code of PLN with the following changes:

- add PLN as a new service type and read mapping service entries from topology files
- define the configuration to start the service
- initialise the service with a messenger and database
- start service

### CODE REVIEW

File: go/pkg/pln/config/config.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	"github.com/scionproto/scion/go/lib/addr"
+	"github.com/scionproto/scion/go/lib/config"
+	"github.com/scionproto/scion/go/lib/env"
+	"github.com/scionproto/scion/go/lib/log"
+	"github.com/scionproto/scion/go/lib/serrors"
+)
+
+const (
+	DefaultDb                = "./pln.db"
+	DefaultPropagateInterval = 1 * time.Hour //1 day
```
Comment (FR4NK-W) [inline go/pkg/pln/config/config.go]: `1 * time.Hour //1 day`

I assume the comment is incorrect and should say 1 hour

File: go/pkg/pln/config/config.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+var _ config.Config = (*PlnConf)(nil)
+
+type PlnConf struct {
+	// ID of the PLN  (required)
+	ID string `toml:"id,omitempty"`
+
+	// DispatcherBypass is the underlay address (e.g. ":30041") to use when bypassing SCION
+	// dispatcher. If the field is empty bypass is not done and SCION dispatcher is used
+	// instead.
+	DispatcherBypass string `toml:"disaptcher_bypass,omitempty"`
```
Comment (FR4NK-W) [inline go/pkg/pln/config/config.go:83]: >`disaptcher_bypass`

dispatcher_bypass

Can you add unit tests for this? I see you copied over the one for the mapping service with `CheckTestMS` in ' go/pkg/pln/config/configtest/configtest.go'.
Unit tests could check that options that are set also get applied.

Comment (FR4NK-W) [reply go/pkg/pln/config/config.go:83]: Can you also check that your existing tests pass, please?
I cannot see any declaration for `CheckTestPLN`, I guess the `CheckTestMS` in the pln config package should be renamed and adapted.

File: go/pkg/pln/config/sample.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+quic_addr = "127.0.0.27:20655" 
+
+#CertFile for QUIC socket (required)
+cert_file = "gen-certs/tls.pem" 
+
+#KeyFile for QUIC socket (required)
+key_file = "gen-certs/tls.key" 
+
+#PropagateInterval is the time interval between PLN list propagations (default = 1 hour)
+prop_interval = 24
```
Comment (FR4NK-W) [inline go/pkg/pln/config/sample.go]: >`(default = 1 hour)

>prop_interval = 24`

Is the propagation interval supposed to be 1 hour or 24 hours? There is the same confusion in other places.

File: go/lib/topology/topology.go [hunk 4]
```diff
@@ -320,6 +327,8 @@ func (t *RWTopology) getSvcInfo(svc proto.ServiceType) (*svcInfo, error) {
 		return &svcInfo{idTopoAddrMap: t.CS}, nil
 	case proto.ServiceType_sig:
 		return &svcInfo{idTopoAddrMap: t.SIG}, nil
+	case proto.ServiceType_pln:
```
Comment (FR4NK-W) [inline go/lib/topology/topology.go:335]: >`case proto.ServiceType_pln:`

You forgot to add `ServiceType_pln` to 'common.capnp.go', the service type is there in 'common.capnp'

**Review by FR4NK-W — CHANGES_REQUESTED** [review]

File: go/pkg/pln/config/config.go (hunk 1 above)
Comment (suprajasridhara) [reply go/pkg/pln/config/config.go]: Yes. That's a mistake

File: go/lib/topology/topology.go (hunk 4 above)
Comment (suprajasridhara) [reply go/lib/topology/topology.go:335]: I thought the capnp.go files are generated when scion build is run. So I didn't commit that change

File: go/pkg/pln/config/sample.go (hunk 3 above)
Comment (suprajasridhara) [reply go/pkg/pln/config/sample.go]: Its 1 hour. I'll change this as well

File: go/pkg/pln/config/config.go (hunk 2 above)
Comment (suprajasridhara) [reply go/pkg/pln/config/config.go:83]: I had adapted it but forgot to rename the method

**Pull request merged**